bundles/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/handler/NibeHeatPumpHandler.java

```diff
@@ -533,25 +514,49 @@ private void handleVariableUpdate(PumpModel pumpModel, ModbusValue value) {
         if (variableInfo != null) {
             logger.trace("Using variable information to register {}: {}", coilAddress, variableInfo);
 
-            double val = (double) value.getValue() / (double) variableInfo.factor;
-            logger.debug("{} = {}", coilAddress + ":" + variableInfo.variable, val);
+            int val = value.getValue();
+            logger.debug("{} = {}", coilAddress + ":" + variableInfo.variable + "/" + variableInfo.factor, val);
 
             CacheObject oldValue = stateMap.get(coilAddress);
-            stateMap.put(coilAddress, new CacheObject(System.currentTimeMillis(), val));
 
-            if (oldValue != null && val == oldValue.value) {
+            if (oldValue != null && val == oldValue.value && (oldValue.lastUpdateTime + refreshIntervalMillis() / 2) >= System.currentTimeMillis()) {
```

Comment:
Why refresh interval is diveded by 2?

Reply:
If I set it to just `oldValue.lastUpdateTime + refreshIntervalMillis()` then binging requests to read variables one by one that are already included in the readout. We do not want that.

BTW. If we do `updateState `on every readout. Nothing is logged to the logs. Core just filters these events out. So @wborn is totally right.

Reply:
Thanks for also checking it! I also retested this last week and didn't see same state updates being sent to frontends using SSE.

Previously channel updates were expensive operations but this bottleneck was fixed with several PRs for https://github.com/eclipse/smarthome/issues/6416. It made channel updates for each binding a lot faster and also allowed for removing update optimizations for each binding. I removed mine from the Nest Binding in https://github.com/openhab/openhab2-addons/pull/4203. :-)

Reply:
@paulianttila So how you want it?

Reply:
@paulianttila Are you fine with current version?

Reply:
I just read this post on the [forum](https://community.openhab.org/t/your-setup-for-a-wall-mounted-tablet/22540/146) where this guy implemented a filter to rest interface and modified habpanel. So that put me in doubt if this problem is really solved.

Reply:
In that post he speaks about problem of initial load of ALL items to habpanel. Not about periodic updates.

What `refreshInterval` you personally use?
Why it is bad to send just **2** updates to unchanged channels (read from readout) during that period?

Reply:
I am running version of this pull request of the binding from end of May. I do not see any lags anywhere. Updates are minimal. events.log get updated only for changes. I control Point Offset with it, depending on the indoor climate, sun activity and weather forecast.
`expires` works perfectly. If I turn off the heat pump for ten minutes or more then all items get nullified... Perfect!